pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -630,14 +630,21 @@ self: super: {
 
   # Make elisp files available at a location where people expect it.
   hindent = (overrideCabal super.hindent (drv: {
+    # 2020-10-19: needs MonadFail fix, see https://github.com/mihaimaruseac/hindent/issues/570
+    src = pkgs.fetchFromGitHub {
+      owner = "mihaimaruseac";
+      repo = "hindent";
+      rev = "085417feebf6f24ec211a05eee6f28c3d7cc1caf";
+      sha256 = "0336k34549ni3faz43fp5bpbgx85v6v4fjcincylwggwylfnnill";
+    };
```

Comment:
Normally we try to apply patches from upstream rather than overwriting `src`.  This so that as soon as it it fixed upstream, the derivation starts failing to build (because the patches no longer apply), so we immediately know that the patches can be removed.

If possible, can you rework this to apply whatever the patches from upstream that are needed for supporting later GHCs?

Reply:
Oh, also, is https://github.com/mihaimaruseac/hindent the new home of `hindent`?  The latest hackage package (https://hackage.haskell.org/package/hindent-5.3.1) lists https://github.com/commercialhaskell/hindent as the home... 

which redirects to https://github.com/mihaimaruseac/hindent when you click on it...  Okay, ignore this question then!

Reply:
@cdepillabout I tried the patch way, but after picking more than 3 prs from GitHub which are all necessary to fix this I gave up because there are still more changes required to fix hindent.

The last upload on hackage was on 2019-06-28T08:35:20Z so there are quite some changes compared to `master` unfortunately...

Reply:
As long as everything has been merged into `master`, you should be able to use a URL like the following:

https://github.com/mihaimaruseac/hindent/compare/5.3.1...75e6f1038dfbbe8ebd9481a5df51586b780a4e74.patch

This generates a patch file between two arbitrary commits/branches/tags.  Is it possible to use something like this to get all the commits you need?

Reply:
That's what I tried, but if fails due to changes to e.g. `travis.yml` files which are not present in the source fetched from hackage :sweat_smile:  Any ideas about that? :grin:

Reply:
Hmm, I vaguely remember that you can pass flags to `fetchpatch` that causes it to ignore certain files.  Maybe that would work??

Reply:
Super tip, I tried that but for some reason the downloaded patch from GitHub for the correct range does not work due to a line being changed twice and leading to patch rejections.

I manually fixed the patch and added it to the patches folder.

Thanks for the hints!